### PR TITLE
feat: support logout

### LINKED ISSUE #32: Support logout
Design discussion: #18 

Proposed API:
```go
package credentials

func Logout(ctx context.Context, store Store, registryName string) error {
	panic("not implemented") // TODO: Implement
}
```

References:

- [Notation logout](https://github.com/notaryproject/notation/blob/main/cmd/notation/logout.go#L50-L53)
- [Oras logout](https://github.com/oras-project/oras/blob/main/cmd/oras/root/logout.go#L64-L71)

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #32

### CODE REVIEW

File: logout_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		args    args
+		wantErr bool
+	}{
+		// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := Logout(tt.args.ctx, tt.args.store, tt.args.registryName); (err != nil) != tt.wantErr {
+				t.Errorf("Logout() error = %v, wantErr %v", err, tt.wantErr)
+			}
```
Comment (Wwwsylvia) [inline logout_test.go:41]: We may need to check if the credential has been deleted or not.

File: logout.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+package credentials
+
+import (
+	"context"
+	"fmt"
+)
+
+func Logout(ctx context.Context, store Store, registryName string) error {
+	if err := store.Delete(ctx, registryName); err != nil {
```
Comment (Wwwsylvia) [inline logout.go:24]: Should we redirecting `docker.io` like oras does? See https://github.com/oras-project/oras/blob/main/cmd/oras/root/logout.go#L64-L70. /cc @qweeah

File: logout_test.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply logout_test.go:41]: Added unit tests for this.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

File: registry.go [hunk 3]
```diff
@@ -55,6 +55,15 @@ func Login(ctx context.Context, store Store, reg *remote.Registry, cred auth.Cre
 	return nil
 }
 
+// Logout provides the logout functionality given the registry name.
+func Logout(ctx context.Context, store Store, registryName string) error {
+	registryName = mapHostname(registryName)
+	if err := store.Delete(ctx, registryName); err != nil {
+		return fmt.Errorf("unable to logout of the registry %s: %w", registryName, err)
```
Comment (shizhMSFT) [inline registry.go:62]: This error message is not consistent with line 53.

BTW, for line 49, it should not be "unable to login to the registry" but "unable to ping the registry".

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: registry.go (hunk 3 above)
Comment (wangxiaoxuan273) [reply registry.go]: changed accordingly.

**Pull request merged**